addons/mass_mailing/models/mass_mailing.py

```diff
@@ -80,102 +64,134 @@ class MassMailingContact(osv.Model):
-        'email': fields.char('Email', required=True),
-        'create_date': fields.datetime('Create Date'),
-        'list_id': fields.many2one(
-            'mail.mass_mailing.list', string='Mailing List',
-            ondelete='cascade', required=True,
-        ),
-        'opt_out': fields.boolean('Opt Out', help='The contact has chosen not to receive mails anymore from this list'),
-        'unsubscription_date': fields.datetime('Unsubscription Date'),
-        'message_bounce': fields.integer('Bounce', help='Counter of the number of bounced emails for this contact.'),
-    }
-
-    def _get_latest_list(self, cr, uid, context={}):
-        lid = self.pool.get('mail.mass_mailing.list').search(cr, uid, [], limit=1, order='id desc', context=context)
-        return lid and lid[0] or False
-
-    _defaults = {
-        'list_id': _get_latest_list
-    }
-
-    def on_change_opt_out(self, cr, uid, id, opt_out, context=None):
-        return {'value': {
-            'unsubscription_date': opt_out and fields.datetime.now() or False,
-        }}
-
-    def create(self, cr, uid, vals, context=None):
+    name = fields.Char()
+    email = fields.Char(required=True)
+    create_date = fields.Datetime(string='Create Date')
+    list_id = fields.Many2one(
+        'mail.mass_mailing.list', string='Mailing List',
+        ondelete='cascade', required=True, default=lambda self: self.env['mail.mass_mailing.list'].search([], limit=1, order='id desc'))
+    opt_out = fields.Boolean(string='Opt Out', help='The contact has chosen not to receive mails anymore from this list')
+    unsubscription_date = fields.Datetime(string='Unsubscription Date')
+    message_bounce = fields.Integer(string='Bounce', help='Counter of the number of bounced emails for this contact.')
+
+    @api.onchange('opt_out')
+    def on_change_opt_out(self):
+        self.unsubscription_date = self.opt_out and fields.Datetime.now()
+
+    @api.model
+    def create(self, vals):
         if 'opt_out' in vals:
-            vals['unsubscription_date'] = vals['opt_out'] and fields.datetime.now() or False
-        return super(MassMailingContact, self).create(cr, uid, vals, context=context)
+            vals['unsubscription_date'] = vals['opt_out'] and fields.Datetime.now() or False
+        return super(MassMailingContact, self).create(vals)
 
-    def write(self, cr, uid, ids, vals, context=None):
+    @api.multi
+    def write(self, vals):
         if 'opt_out' in vals:
-            vals['unsubscription_date'] = vals['opt_out'] and fields.datetime.now() or False
-        return super(MassMailingContact, self).write(cr, uid, ids, vals, context=context)
+            vals['unsubscription_date'] = vals['opt_out'] and fields.Datetime.now() or False
+        return super(MassMailingContact, self).write(vals)
 
-    def get_name_email(self, name, context):
-        name, email = self.pool['res.partner']._parse_partner_name(name, context=context)
+    def get_name_email(self, name):
```

Comment:
By removing the context, you'll have some trouble with this route in website_mass_mailing
``` python
    @http.route('/website_mass_mailing/subscribe', type='json', website=True, auth="public")
    def subscribe(self, list_id, email, **post):
        cr, uid, context = request.cr, request.uid, request.context
        Contacts = request.registry['mail.mass_mailing.contact']
        parsed_email = Contacts.get_name_email(email, context=context)[1]
```